### PR TITLE
Typescript improvements and Address Checksumming

### PR DESCRIPTION
**Pull request by mickys**
### 1. Cleanup and Typescript improvements.
- Cleaned up repository of .lock files
- Added util, schnorr, znode, exports to index for easy import
- Added prepare for /dist generation upon install so types can actually be used in TypeScript

#### 1.1. New usage example no longer requires "Zilliqa" class instantiation to get to the util library
```typescript
import { util } from 'zilliqa-js';

util.isAddress( "not_an_address" ); // false
util.isAddress( "7bb3b0e8a59f3f61d9bff038f4aeb42cae2ecce8" ); // true
```

### 2. Address Checksumming based on EIP-55 
**see: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md**

#### 2.1. Methods:
**methods support both standard and 0x prepended addresses**
- toChecksumAddress(address: string): string
- isValidChecksumAddress(address: string): boolean

#### 2.2. Usage Example
```typescript
import { util } from 'zilliqa-js';

const address = "0x7bb3b0e8a59f3f61d9bff038f4aeb42cae2ecce8";
const checksummed = util.toChecksumAddress( address );
// 0x7bB3b0E8A59f3f61d9BFf038f4AEB42Cae2eccE8

util.isValidChecksumAddress( checksummed ); // true
util.isValidChecksumAddress( address ); // false
```

**methods support both standard and 0x prepended addresses as inputs**
```typescript
const lowercaseAddress = "7bb3b0e8a59f3f61d9bff038f4aeb42cae2ecce8";

util.toChecksumAddress( lowercaseAddress );
// 0x7bB3b0E8A59f3f61d9BFf038f4AEB42Cae2eccE8

util.toChecksumAddress( "0x" + lowercaseAddress );
// 0x7bB3b0E8A59f3f61d9BFf038f4AEB42Cae2eccE8

util.isValidChecksumAddress( "0x7bB3b0E8A59f3f61d9BFf038f4AEB42Cae2eccE8".replace('0x', '') );
// true
util.isValidChecksumAddress( "0x7bB3b0E8A59f3f61d9BFf038f4AEB42Cae2eccE8" );
// true

```

#### 2.3. Tests
Added tests for these in __tests__/util.spec.ts

## Review Suggestion
Checksumming uses: hashjs.sha256() since it was already imported in the util class

### Implementation
- [X] **ready for review**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
   "scripts": {
     "build": "rimraf dist/* && tsc --emitDeclarationOnly && cross-env NODE_ENV=production webpack --config webpack.config.js",
     "test": "jest -c jest.config.js",
-    "publish": "yarn build && yarn publish"
+    "publish": "yarn build && yarn publish",
+    "prepare": "npm run build"
```
[the remark this answers is not in the document]
Comment (mickys) [reply package.json]: If the package is used as a dependency in a project, we should be able to use the types provided in the dist folder.

"prepare" tells the npm installer to run a build once the package is installed, in order to generate the dist directory.

**Comment by iantanwx** [comment]
Thanks @mickys. This looks good. Would you be able to add a few more test vectors for the checksum tests? You can do something like `checksum.fixtures.ts`.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (iantanwx) [reply package.json]: @mickys you're right. It looks like there has been some mistake when the build was published. We'll rectify this issue. `prepare` should not be necessary.

@edisonljh which folder do we normally publish?

[the remark this answers is not in the document]
Comment (mickys) [reply package.json]: I've removed the prepare instruction from the PR.
Feel free to merge then do the changes required.

**Comment by mickys** [comment]
I've removed the prepare instruction from the PR.
Feel free to merge then do the types required changes you're planning.

**Comment by edisonljh** [comment]
This looks good! Thanks, this is a good PR! 👍

**Comment by iantanwx** [comment]
Thanks @mickys !

**Pull request merged**